WcaOnRails/app/jobs/compute_auxiliary_data.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class ComputeAuxiliaryData < TimedApplicationJob
+  queue_as :default
+
+  def perform
+    # Note: During the results posting process some results may be missing their corresponding WCA ID.
+    #       If we detect that (means someone else is posting results at the moment), we defer the computation.
```

Comment:
I also imagine it's fairly common that we're in the middle of results submission without any empty WCA IDs. If the competition had no first time competitors is one case.

If it's not a big deal anyway, that's fine, I guess.